### PR TITLE
fix(ws): Add readonly modifier and some other refactors and format

### PR DESCRIPTION
**Pull request by seahindeniz**
I've noticed some properties doesn't have `readonly` restriction and literal types, so I added them.
Instead of importing entire contents of modules, I replaced them with only the ones that needed in the types.
Declaration file has formatted with Prettier as well.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/constants.js#L4>>

### CODE REVIEW

File: types/ws/index.d.ts [hunk 1]
```diff
@@ -12,134 +12,171 @@
...
+    static readonly CLOSING: 2;
     /** The connection is closed. */
-    static CLOSED: 3;
+    static readonly CLOSED: 3;
 
-    binaryType: string;
-    bufferedAmount: number;
-    extensions: string;
-    protocol: string;
+    binaryType: "nodebuffer" | "arraybuffer" | "fragments";
```
Comment (seahindeniz) [inline types/ws/index.d.ts:42]: Please see here for declared binary types https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/constants.js#L4

File: types/ws/index.d.ts [hunk 2]
```diff
@@ -12,134 +12,171 @@
...
+    readonly bufferedAmount: number;
+    readonly extensions: string;
```
Comment (seahindeniz) [inline types/ws/index.d.ts:44]: https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/websocket.js#L111-L113

File: types/ws/index.d.ts [hunk 3]
```diff
@@ -12,134 +12,171 @@
...
+    readonly protocol: string;
```
Comment (seahindeniz) [inline types/ws/index.d.ts:45]: https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/websocket.js#L118-L120

File: types/ws/index.d.ts [hunk 4]
```diff
@@ -12,134 +12,171 @@
...
     /** The current state of the connection */
-    readyState:
+    readonly readyState:
```
Comment (seahindeniz) [inline types/ws/index.d.ts:47]: https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/websocket.js#L125-L127

File: types/ws/index.d.ts [hunk 5]
```diff
@@ -12,134 +12,171 @@
...
         | typeof WebSocket.CONNECTING
         | typeof WebSocket.OPEN
         | typeof WebSocket.CLOSING
         | typeof WebSocket.CLOSED;
-    url: string;
+    readonly url: string;
```
Comment (seahindeniz) [inline types/ws/index.d.ts:52]: https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/websocket.js#L132-L134

File: types/ws/index.d.ts [hunk 6]
```diff
@@ -12,134 +12,171 @@
...
+    static readonly CONNECTING: 0;
     /** The connection is open and ready to communicate. */
-    static OPEN: 1;
+    static readonly OPEN: 1;
     /** The connection is in the process of closing. */
-    static CLOSING: 2;
+    static readonly CLOSING: 2;
     /** The connection is closed. */
-    static CLOSED: 3;
+    static readonly CLOSED: 3;
```
Comment (seahindeniz) [inline types/ws/index.d.ts:40]: https://github.com/websockets/ws/blob/114de9e33668075f0af88dc440f1ebd813161e72/doc/ws.md#ready-state-constants
https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/websocket.js#L26

**Comment by seahindeniz** [comment]
RIP everyone 😃

**Review by reduckted — APPROVED** [review]

**Comment by seahindeniz** [comment]
Ready to merge

**Pull request merged**